src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1212,7 +1212,7 @@ private TypeWithAnnotations BindGenericSimpleNamespaceOrTypeOrAliasSymbol(
 
             if (isUnboundTypeExpr)
             {
-                if (!IsUnboundTypeAllowed(node))
+                if (!IsInsideNameof && !IsUnboundTypeAllowed(node))
```

Comment:
From my understanding, we want `nameof` to behave just like `typeof` with regards to open types.
So it feels like the change should be inside an override of `IsUnboundTypeAllowed` in `NameofBinder` instead, implemented in a similar fashion to `TypeofBinder.IsUnboundTypeAllowed`.

Then depending on where the LangVersion lands, we may not need any change to `ConstructNamedTypeUnlessTypeArgumentOmitted` below and we may not need dedicated validation logic (`validateOpenGenericNames` and `ERR_NestedUnboundTypeNotAllowedInNameofExpression`).